src/script/script.js

```diff
@@ -0,0 +1,402 @@
+    this.rootElement.remove();
+  }
+
+  ///////////////////////////////////////////////////////////////
+  switchLanguage() {
+    localStorage.keyboardLanguage = localStorage.keyboardLanguage === "ru" ? "en" : "ru";
+    for (let key of this.keys) {
+      let items = key.children;
+      Array.from(items).forEach(el => {
+        if (el.classList.contains(localStorage.keyboardLanguage)) {
+          el.classList.remove('hidden');
+          this.toggleClassesInButton(key, "caseDown");
+        } else {
+          el.classList.add('hidden');
+        }
+      });
+    }
+  }
+
+  textAreaChangeInfo(value) {
+    let text = this.textArea.value;
+    this.textArea.focus();
+    if (value === "&amp;"){
+      this.textArea.setRangeText("&", this.textArea.selectionStart, this.textArea.selectionEnd, "end");
+    } else if (value === "Backspace") {
+      if (this.currentPosition) {
+        this.textArea.value = text.slice(0, this.currentPosition - 1) + '' + text.slice(this.currentPosition, text.length);
+        this.textArea.setRangeText("", this.currentPosition, this.currentPosition, "end");
+      } else {
+        this.textArea.value = text.slice(0, -1);
+      }
+    } else if (value === "Delete"){
+      this.textArea.value = text.slice(0, this.currentPosition) + '' + text.slice(this.currentPosition+1, text.length);
+      this.textArea.setRangeText("", this.currentPosition, this.currentPosition, "end");
+    } else {
+      this.textArea.setRangeText(value, this.textArea.selectionStart, this.textArea.selectionEnd, "end");
+    }
+  }
+
+  handleButtonDown(eventCode) {
+    if (this.exceptions.includes(eventCode)) return;
+    let clickedButton = document.querySelector(`.${eventCode}`);
+    if (eventCode === "CapsLock") {
+      clickedButton.classList.toggle('active');
+      this.capslocked = !this.capslocked;
+      this.toggleCappslock();
+      return;
+    }
+    if (!this.pressed.has(eventCode)) {
+      this.pressed.add(eventCode);
+      clickedButton.classList.toggle('active');
+    }
+
+    if (clickedButton.classList.contains('symbol')) {
+      this.textAreaChangeInfo(clickedButton.dataset.value);
+    } else {
+      if (eventCode === "ShiftRight" || eventCode === "ShiftLeft") {
+        this.toggleShift();
+        return;
+      }
+      if (eventCode === "ControlLeft" || eventCode === "AltLeft") {
+        for (let code of this.codes) {
+          if (!this.pressed.has(code)) {
+            return;
+          }
+          this.switchLanguage();
+        }
+        return;
+      }
+      if (eventCode === "Space") {
+        this.textAreaChangeInfo(" ");
+        return;
+      }
+      if (eventCode === "ContextMenu") {
+        this.switchLanguage();
+        return;
+      }
+      if (eventCode === "Enter") {
+        this.textAreaChangeInfo("\n");
+        return;
+      }
+      if (eventCode === "Tab") {
+        this.textAreaChangeInfo("\t");
+        return;
+      }
+      if (eventCode === "Backspace") {
```

Comment:
Unnessasary code here and further.
Think how to avoid code duplication

Reply:
Fixed. Replaced to:
```suggestion
      switch (eventCode) {
        case 'Space':
          this.textAreaChangeInfo(' ');
          break;
        case 'ContextMenu':
          this.switchLanguage();
          break;
        case 'Enter':
          this.textAreaChangeInfo('\n');
          break;
        case 'Tab':
          this.textAreaChangeInfo('\t');
          break;
        case 'Backspace':
          this.textAreaChangeInfo('Backspace');
          break;
        case 'Delete':
          this.textAreaChangeInfo('Delete');
          break;
        case 'ArrowLeft':
          this.textAreaChangeInfo('◄');
          break;
        case 'ArrowDown':
          this.textAreaChangeInfo('▼');
          break;
        case 'ArrowRight':
          this.textAreaChangeInfo('►');
          break;
        case 'ArrowUp':
          this.textAreaChangeInfo('▲');
          break;
        default:
          break;
      }
```